go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,221 @@
+		// Also, if it is not an infra container, nothing to do in this case as well.
+		logrus.Infof("CNI called for normal container or infra container, nothing to do")
+		return []*current.Interface{}, nil
+	}
+
+	ipAddr, ipNet, err := net.ParseCIDR(ipAddress)
+	if err != nil {
+		return nil, err
+	}
+	ipMaskSize, _ := ipNet.Mask.Size()
+	endpointName := args.ContainerID
+
+	var hnsNetworkId string
+	hnsNetworkId, err = getHNSIdFromConfigOrByGatewayIP(gatewayIP)
+	if err != nil {
+		logrus.Infof("Error when detecting the HNS Network Id: %q", err)
+		return nil, err
+	}
+
+	// Ensure that the macAddress is given in xx:xx:xx:xx:xx:xx format
+	macAddressIpFormat := strings.Replace(macAddress, ":", "-", -1)
+
+	// Check if endpoint is created, otherwise create it.
+	// This is to make the call to add idempotent
+	var createdEndpoint *hcsshim.HNSEndpoint
+	createdEndpoint, err = hcsshim.GetHNSEndpointByName(endpointName)
+	if err != nil {
+		logrus.Infof("HNS endpoint %q does not exist", endpointName)
+		hnsEndpoint := &hcsshim.HNSEndpoint{
+			Name:           endpointName,
+			VirtualNetwork: hnsNetworkId,
+			IPAddress:      ipAddr,
+			MacAddress:     macAddressIpFormat,
+			PrefixLength:   uint8(ipMaskSize),
+		}
+		createdEndpoint, err = createHNSEndpoint(hnsEndpoint)
+		if err != nil {
+			logrus.Warningf("HNS endpoint failed to create, reason: %q", err)
+			return nil, err
+		}
+	} else {
+		logrus.Infof("HNS endpoint already exists with name: %q", endpointName)
+	}
+
+	err = containerHotAttachEndpoint(createdEndpoint, args.ContainerID)
+	if err != nil {
+		logrus.Warningf("Failed to hot attach HNS Endpoint %q to container %q, reason: %q", endpointName, args.ContainerID, err)
+		return nil, err
+	}
+
+	ifaceID := fmt.Sprintf("%s_%s", namespace, podName)
+	// "--may-exist"  is added to support the function idempotency
+	ovsArgs := []string{
+		"--may-exist", "add-port", "br-int", endpointName, "--", "set",
+		"interface", endpointName, "type=internal", "--", "set",
+		"interface", endpointName,
+		fmt.Sprintf("external_ids:attached_mac=%s", macAddress),
+		fmt.Sprintf("external_ids:iface-id=%s", ifaceID),
+		fmt.Sprintf("external_ids:ip_address=%s", ipAddress),
+	}
+	var out []byte
+	out, err = exec.Command("ovs-vsctl", ovsArgs...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failure in plugging pod interface: %v  %q", err, string(out))
+	}
+
+	return []*current.Interface{}, nil
+}
+
+func PlatformSpecificCleanup(args *skel.CmdArgs) error {
+	endpointName := args.ContainerID
+	ovsArgs := []string{
+		"del-port", "br-int", endpointName,
+	}
+	out, err := exec.Command("ovs-vsctl", ovsArgs...).CombinedOutput()
+	if err != nil && !strings.Contains(string(out), "no port named") {
+		// DEL should be idempotent; don't return an error just log it
+		logrus.Warningf("failed to delete OVS port %s: %v  %q", endpointName, err, string(out))
+	}
+	return deleteHNSEndpoint(endpointName)
```

Comment:
`func cmdDel` always returned nil (on Linux), do we need to worry about returning an error here?

Reply:
Added a comment, if an error is returned it means the HNS Endpoint could not be deleted. Ignoring the error might lead to having leaked HNS Endpoints and we don't want that.